livekit-agents/livekit/agents/tts/tts.py

```diff
@@ -112,6 +124,7 @@ def flush(self) -> None:
         self._check_input_not_ended()
         self._check_not_closed()
         self._input_ch.send_nowait(self._FlushSentinel())
+        self._pending += 1
```

Comment:
What is _pending used for?

Reply:
every flush pending gets incremented, and every response it gets decremented. if pending is 0, then we're not waiting for new data, and we can wait forever. if pending is positive, then something is processing, and we need to time out after a while.

Reply:
an alternative would be to have another channel, that gets input on each flush:

```python
    def flush(self) -> None:
        """Mark the end of the current segment"""
        self._check_input_not_ended()
        self._check_not_closed()
        self._input_ch.send_nowait(self._FlushSentinel())
        self._req_ch.send_nowait(None)

    async def __anext__(self) -> SynthesizedAudio:
        async for _ in self._req_ch:
            return await asyncio.wait_for(
                self._event_ch.__anext__(), self._timeout
            )
```

this is cleaner, actually. let me know if it looks good and i'll implement it

Reply:
One thing to note is that a flush can still be done even if no data was pushed

Reply:
this doesn't matter, as long as the ratio of flushes to expected events is 1:1.

Reply:
we can move the `_req_ch.send_nowait` to the plugin side, to only send if FlushSentinel was sent after data and not by itself